TU20Bot/Configuration/AuthorizationModule.cs

```diff
@@ -0,0 +1,186 @@
+        /// </summary>
+        /// <param name="jwtToken"></param>
+        /// <returns>AuthorizationPayload instance from the JWT token</returns>
+        /// <throws>Exception</throws>
+        public static AuthorizationPayload attemptValidateToken(string jwtToken) {
+
+            if (string.IsNullOrEmpty(jwtToken)) {
+                throw new Exception("ERROR: Empty payload.");
+            }
+
+            // Deserialize the JWT Token into an AuthorizationPayload object
+            var payload = JsonConvert.DeserializeObject<AuthorizationPayload>(jwtToken);
+
+            if (payload == null) {
+                throw new Exception("ERROR: Invalid payload.");
+            }
+
+            // If the token is expired
+            if (DateTime.Now.CompareTo(payload.validUntil) > 0) {
+                throw new Exception("ERROR: Expired token.");
+            }
+
+            return payload;
+        }
+
+        /// <summary>
+        /// Check if the user of the current session is signed in and if they contain the appropriate permissions
+        /// </summary>
+        /// <param name="permissions"></param>
+        /// <param name="request"></param>
+        /// <param name="response"></param>
+        /// <param name="jwtSecret"></param>
+        /// <returns></returns>
+        public static bool validatePermissions(List<string> permissions, string httpHeader, string jwtSecret) {
+            if (tokenExists(httpHeader) != null) return false;
+
+            var authorizationHeader = httpHeader;
+
+            // Extract the incoming JWT Token from the header
+            var token = authorizationHeader[bearerPrefix.Length..];
+
+            var secret = Encoding.UTF8.GetBytes(jwtSecret);
+
+            try {
+                var result = JWT.Decode(token, secret);
+                var payload = JsonConvert.DeserializeObject<AuthorizationPayload>(result);
+
+                // Return if the payload contains the desired permission information
+                return payload.permissions.Any(_permission => permissions.Contains(_permission));
+            } catch {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Same as `AuthorizationModule#validatePermissions(List<string> permissions, string httpHeader, string jwtSecret)`,
+        ///     except it additionally writes and flushes errors to httpResponse.
+        /// </summary>
+        /// <param name="permissions"></param>
+        /// <param name="header"></param>
+        /// <param name="jwtSecret"></param>
+        /// <param name="httpResponse"></param>
+        /// <returns></returns>
+        public static bool validatePermissions(List<string> permissions, string header, string jwtSecret, IHttpResponse httpResponse) {
+            var response = tokenExists(header);
+
+            if (response != null) {
+                // Write an error to the HTTP stream
+                error(response, httpResponse);
```

Comment:
I would drop the response parameter and throw or `return 'error message'` instead.